src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -546,13 +551,12 @@ public int MaxResponseDrainSize
 
         protected override void Dispose(bool disposing)
         {
-            if (disposing && !_disposed)
+            if (!_disposed)
```

Comment:
I'm not sure how this is working today since the callback is already configured for the `requestHandle`. 
According to [Handle Hierarchy](https://msdn.microsoft.com/en-us/library/windows/desktop/aa383880(v=vs.85).aspx#_hierarchy):

> After closing a handle, the application must be prepared to receive callback notifications on the handle until the final `WINHTTP_CALLBACK_STATUS_HANDLE_CLOSED` value is returned to indicate that the handle is completely closed.

Shouldn't the AppDomain be protected from disposing by leaving at least an object pinning it?
The WinHttpResponseStream Dispose will set all references to the handle to null. I believe this means that there's nothing to pin the AppDomain object (and thus the callback) from getting GCed prior to the final callback from WinHttp.